augur/generate.py

```diff
@@ -0,0 +1,266 @@
+#
+# Data generation module
+#
+
+import firecrown
+import sacc
+import copy
+import numpy as np
+
+
+def generate(config):
+    """ Generates data using dictionary based config.
+
+    Parameters:
+    ----------
+    config : dict
+        The yaml parsed dictional of the input yaml file
+
+    """
+
+    verbose = config['verbose']
+    gen_config = config['generate']
+    gen_config['verbose'] = verbose
+
+    capabilities = [
+        ('two_point',
+         'firecrown.ccl.two_point',
+         two_point_template,
+         two_point_insert)]
+    process = []
+
+    for dname, ddict in gen_config.items():
+        if (not hasattr(ddict, "__getitem__") or
```

Comment:
pls add a comment about what exactly you are looking for here. Also, you might consider using the `collections.abc.MutableMapping` type for this test with `isinstance`. It depends on what exactly you want here.